Source/CesiumRuntime/Public/CesiumMetadataPrimitive.h

```diff
@@ -48,10 +48,16 @@ struct CESIUMRUNTIME_API FCesiumMetadataPrimitive {
       const CesiumGltf::ExtensionMeshPrimitiveExtFeatureMetadata&
           primitiveMetadata);
 
+  const CesiumGltf::Model& GetModel() const;
+  const CesiumGltf::MeshPrimitive& GetPrimitive() const;
+
 private:
   TArray<FCesiumMetadataFeatureTable> _featureTables;
   VertexIDAccessorType _vertexIDAccessor;
 
+  CesiumGltf::Model _model;
+  CesiumGltf::MeshPrimitive _meshPrimitive;
```

Comment:
Holding the CesiumGltf::Model by value means that the entire model - including all its data buffers - will be copied into the FCesiumMetdataPrimitive at construction, which is going to be really expensive. A better alternative is to hold (const) pointers to these objects.